plugins/rmanipulation/CMakeLists.txt

```diff
@@ -7,13 +7,10 @@ add_library(rmanipulation SHARED rmanipulation.cpp basemanipulation.cpp    plugi
 if( Boost_REGEX_FOUND )
   message(STATUS "boost regex found")
   set(PLUGIN_COMPILE_FLAGS "${PLUGIN_COMPILE_FLAGS} -DHAVE_BOOST_REGEX")
-  target_link_libraries(rmanipulation libopenrave ${Boost_REGEX_LIBRARY})
 else()
   message(STATUS "failed to find boost regex, please install it")
-  target_link_libraries(rmanipulation libopenrave)
 endif()
-target_link_libraries(rmanipulation PRIVATE boost_assertion_failed)
-
+target_link_libraries(rmanipulation PRIVATE boost_assertion_failed PUBLIC libopenrave $<$<BOOL:Boost_REGEX_FOUND>:${Boost_REGEX_LIBRARY}>)
```

Comment:
@undisputed-seraphim what's the point of making this harder to read?

Reply:
It's... really not harder to read? Well, I guess I'll revert it if it's too hard.